### PR TITLE
Major updates of Command Line Interface

### PR DESCRIPTION
**Pull request by andylvua**
## CLI update
* Now screens are using convenient options that can be defined using implicit lambda expressions or macro definitions
* CLI now properly using shared pointers for navigation
* CLI has been moved to use the `ncurses` library, which increases portability and facilitates further development

### Additional updates:
* Raspberry-specific compilation targets to improve cross-platform compatibility. No need for restructuring the code every time 
* Debug flags. Possibility to exclude the initialization of the module during the period of active development in order to speed up the loading of the program

### CODE REVIEW

**Review by codefloww — COMMENTED** [review]
forgot to include functional.h in option.h

**Review by codefloww — COMMENTED** [review]
missing some rerendering in leaf screens

**Review by codefloww — COMMENTED** [review]
io not working for reading number or message in leaf screens

**Comment by andylvua** [comment]
> forgot to include functional.h in option.h

Fixed in 07703452fc31c75c92dd08cafebeba8e27ec8d34



> missing some rerendering in leaf screens

> io not working for reading number or message in leaf screens

Fixed in 42db2c9a7ebee3f13c7b417170d5b9e697584b49, f0db8a0677d530ab68e5f49429ea6e5c09d325af

File: CMakeLists.txt [hunk 1]
```diff
@@ -37,4 +50,4 @@ endif()
 target_link_libraries(phone spdlog::spdlog)
 
 set(CMAKE_PREFIX_PATH "/usr/local/bin")
-target_link_libraries(phone wiringPi)
\ No newline at end of file
+#target_link_libraries(phone wiringPi)
```
Comment (codefloww) [inline CMakeLists.txt]: maybe put wiringPi in raspberry build

File: Src/cli/cli.cpp [hunk 2]
```diff
@@ -43,449 +47,356 @@ bool checkNumber(std::string &number) {
...
+
+        int color = (i == static_cast<size_t>(activeOptionIndex)) ? FILLED_WHITE : WHITE;
+        printColored(color, option.toStdString());
     }
 }
 
-void CLI::changeScreen(std::shared_ptr<Screen> screen) {
-    currentScreen = std::move(screen);
+void CLI::changeScreen(const QString &screenName) {
+    qDebug() << "Changing screen to " << screenName;
```
Comment (codefloww) [inline Src/cli/cli.cpp]: remove qdebug pleaseeee

**Review by codefloww — CHANGES_REQUESTED** [review]

**Review by codefloww — COMMENTED** [review]
nice

File: CMakeLists.txt (hunk 1 above)
Comment (andylvua) [reply CMakeLists.txt]: Fixed in 8ebd073636dd55bdc113567c80c63849079ff53c

File: Src/cli/cli.cpp (hunk 2 above)
Comment (andylvua) [reply Src/cli/cli.cpp]: Fixed in 3812eeba4db3f395c3738a1ce1a7c82a5aa25d54

**Pull request merged**